### PR TITLE
Pagination bar fixes

### PR DESCRIPTION
**Pull request by ad9242**
## Summary of the changes
Fixes some issues spotted in storybook docs. This has also led to some prop name changes, but these are not deemed breaking changes as the component is still in canary
Updates Cypress tests
Updates spec test and storybook docs

### General 

- [x] Changes to docs package checked and committed.
- [x] All acceptance criteria reviewed and met. 

### Testing

- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 

### Accessibility 

- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 

### Resize/zoom behaviour 

- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### System modes

- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content.
- [x] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] All prop combinations work without issue. 
- [x] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.
- [x] Controlled and uncontrolled input components tested.

### CODE REVIEW

**Review by GCHQ-Developer-299 — APPROVED** [review]

File: packages/canary-react/src/stories/ic-pagination-bar.stories.mdx [hunk 1]
```diff
@@ -135,31 +135,31 @@ const PaginationBar = () => (
...
+### Pagination range label type
 
-The pagination type can be set to two types:
+The pagination range label type can be set to two types:
 
 - `simple`: [default]: Display the total number of pages and the current page number.
 - `data`: Displays the total number of items and the items range.
 
-The example belows uses the `data` pagination type. The range is updated when the page is changed.
+The example belows uses the `data` pagination range label type. The range is updated when the page is changed.
```
Comment (jd239) [inline packages/canary-react/src/stories/ic-pagination-bar.stories.mdx:145]:
```suggestion
The example below uses the `data` pagination range label type. The range is updated when the page is changed.
```

File: packages/canary-react/src/stories/ic-pagination-bar.stories.mdx [hunk 2]
```diff
@@ -369,27 +397,31 @@ const PaginationBar = () => (
 export default PaginationBar;
 ```
 
-### Hide items per page label
+### Hide range label
 
-If the pagination bar becomes clutted due to the number of pagination elements displayed, the items per page label can be hidden by setting the `hideItemsPerPageLabel` prop to `true`.
+If the pagination bar becomes clutted due to the number of pagination elements displayed, the items per page label can be hidden by setting the `hideRangeLabel` prop to `true`.
```
Comment (jd239) [inline packages/canary-react/src/stories/ic-pagination-bar.stories.mdx:402]:
```suggestion
If the pagination bar becomes clutted due to the number of pagination elements displayed, the range label can be hidden by setting the `hideRangeLabel` prop to `true`.
```

File: packages/canary-web-components/src/components/ic-pagination-bar/ic-pagination-bar.stories.mdx [hunk 3]
```diff
@@ -111,81 +111,101 @@ To display the go to page control, set the `show-go-to-page-control` attribute t
...
+### Pagination range label type
 
-The pagination type can be set to two types:
+The pagination range label type can be set to two types:
 
 - `simple`: [default]: Display the total number of pages and the current page number.
 - `data`: Displays the total number of items and the items range.
 
-The example belows uses the `data` pagination type. The range is updated when the page is changed.
+The example belows uses the `data` pagination range label type. The range is updated when the page is changed.
```
Comment (jd239) [inline packages/canary-web-components/src/components/ic-pagination-bar/ic-pagination-bar.stories.mdx:121]:
```suggestion
The example below uses the `data` pagination range label type. The range is updated when the page is changed.
```

File: packages/canary-web-components/src/components/ic-pagination-bar/ic-pagination-bar.tsx [hunk 4]
```diff
@@ -71,7 +71,7 @@ export class PaginationBar {
   /**
    * The label which will be used in place of 'items' if type is data. Should be capitalised.
    */
-  @Prop() itemLabel?: string = "Item";
+  @Prop() rangeItemLabel?: string = "Item";
```
Comment (jd239) [inline packages/canary-web-components/src/components/ic-pagination-bar/ic-pagination-bar.tsx:74]: I keep on reading this prop in my head and wondering if it should be `range-label-item / rangeLabelItem`? Will let you decide...

**Review by jd239 — CHANGES_REQUESTED** [review]
Checked if adding the 'BREAKING CHANGE' message via Commitizen updates the major version number and it doesn't due to it being a prerelease so think it will be good to add a note of the breaking change in the commit message.

**Review by jd239 — APPROVED** [review]

**Pull request merged**